### DISCUSSION TITLE
Dropdown fields in menu cause error when pushing to production repo

### DISCUSSION THREAD

**Discussion by MoBosse** [open-sdg/open-sdg#1805] [Q&A] [answered]
When I try to push my staging repository (remote theme open-sdg 2.1.0-dev) to the production repo I get an error, saying something like `linking to internal hash # that does not exist` 
See failing test [here](https://github.com/G205SDGs/sdg-indicators/runs/8200724337?check_suite_focus=true)

It seems that this error has something to do with the dropdown fields in the menu. Although it worked in the staging environment, I could not update my production repository until I cleared all the dropdown fields from the menu.
Does anyone else have dropdown items like [here](https://open-sdg.readthedocs.io/en/latest/configuration/#menu) and problems transferring them to the production environment?

**Comment by MoBosse** [comment]
@brockfanning may I ask you to take a look at this? 
I tried to add dropdown menus in a fork of the site-starter repo and had no problems pushing them to the production repo, so this is not a general open-sdg issue.
However when I add dropdown menu items to my actual [site repository](https://github.com/G205SDGs/sdg-indicators), the PRs to the production environment keep [failing](https://github.com/G205SDGs/sdg-indicators/actions/runs/3081506166/jobs/4980125644).

I was wondering if it might have somthing to do with the [html_proofer files](https://github.com/G205SDGs/sdg-indicators/tree/develop/scripts/test), but whatever I changed there also caused failing tests...

**Comment by brockfanning** [comment, marked as answer]
@MoBosse I wonder if it would fix it to change [this line](https://github.com/G205SDGs/sdg-indicators/blob/develop/scripts/test/html_proofer_prod.sh#L4) from...

`bundle exec htmlproofer --disable-external ./_site`

...to...

`bundle exec htmlproofer --allow-hash-href --disable-external ./_site`

If that works I think we need to update the site starter's production script.

**Reply by MoBosse** [reply]
Thanks @brockfanning, 
I have already tried that, but the error remains the same.

**Reply by brockfanning** [reply]
Hi @MoBosse I think it needs to be added to this file: https://github.com/G205SDGs/sdg-indicators/blob/develop/scripts/test/html_proofer_staging.sh

That is the script that is running for each pull-request.

**Comment by MoBosse** [comment]
Yep, that works. Thanks a million!
